pkgs/build-support/setup-hooks/compress-man-pages.sh

```diff
@@ -10,14 +10,15 @@ compressManPages() {
 
     # Compress all uncompressed manpages.  Don't follow symlinks, etc.
     find "$dir"/share/man/ -type f -a '!' -regex '.*\.\(bz2\|gz\|xz\)$' -print0 \
-        | while IFS= read -r -d $'\0' f
+        | xargs -0 -n 128 -P "$NIX_BUILD_CORES" -- \
+            "$SHELL" -c 'for f in "$@"
     do
         if gzip -c -n "$f" > "$f".gz; then
             rm "$f"
         else
             rm "$f".gz
         fi
-    done
+    done' --
```

Comment:
This looks odd

Reply:
The intention is to terminate option processing for the shell. The path names are passed as arguments by `xargs` and without `--` arguments with a leading `-` would be interpreted as options.